packages/snap-url-manager/src/Translators/Url/UrlTranslator.ts

```diff
@@ -148,13 +148,18 @@ export class UrlTranslator implements Translator {
 			.split('&')
 			.filter((v) => v)
 			.map((kvPair) => {
-				const [key, value] = kvPair.split('=').map((v) => decodeURIComponent(v.replace(/\+/g, ' ')));
-				return { key: key.split('.'), value, type: ParamLocationType.query };
+				try {
+					const [key, value] = kvPair.split('=').map((v) => decodeURIComponent(v.replace(/\+/g, ' ')));
+					return { key: key.split('.'), value, type: ParamLocationType.query };
+				} catch (err) {
+					console.error('Uncaught URIError: URI malformed', kvPair);
```

Comment:
Maybe a warning instead of error? Also some context around the error could be helpful:
```
console.warn('Snap UrlTranslator: URI malformed - ignoring parameter', kvPair);
```